code/datums/brain_damage/special.dm

```diff
@@ -182,8 +182,71 @@
 	owner.add_trait(TRAIT_SIXTHSENSE, TRAUMA_TRAIT)
 	active = TRUE
 	addtimer(CALLBACK(src, .proc/cease_whispering), rand(50, 300))
-	
+
 /datum/brain_trauma/special/death_whispers/proc/cease_whispering()
 	owner.remove_trait(TRAIT_SIXTHSENSE, TRAUMA_TRAIT)
 	active = FALSE
 
+/datum/brain_trauma/special/beepsky
+	name = "Criminal"
+	desc = "Patient seems to be a criminal."
+	scan_desc = "criminal mind"
+	gain_text = "<span class='warning'>Justice is coming for you.</span>"
+	lose_text = "<span class='notice'>You were absolved for your crimes.</span>"
+	var/obj/effect/hallucination/simple/securitron/beepsky
+
+/datum/brain_trauma/special/beepsky/on_gain()
+	create_securitron()
+	..()
+
+/datum/brain_trauma/special/beepsky/proc/create_securitron()
+	var/turf/where = locate(owner.x + pick(-12, 12), owner.y + pick(-12, 12), owner.z)
+	beepsky = new(where, owner)
+	beepsky.victim = owner
+
+/datum/brain_trauma/special/beepsky/on_lose()
+	QDEL_NULL(beepsky)
+	..()
+
+/datum/brain_trauma/special/beepsky/on_life()
+	if(get_dist(owner, beepsky) >= 10 && prob(20))
+		QDEL_NULL(beepsky)
+		create_securitron()
+	if(owner.stat != CONSCIOUS)
+		if(prob(20))
+			playsound(owner, 'sound/voice/beepsky/iamthelaw.ogg', 50)
+		return
+	if(!beepsky || !beepsky.loc || beepsky.z != owner.z)
```

Comment:
if(QDELEDED(beekspy)

Reply:
wouldnt that cause problems if you change z level?